TestMurmur3H1 and TestMurmur3H1_CassandraSign fail on s390x

gocql git tip on Fedora Rawhide, Go 1.14 beta 1, on s390x arch only:

```
Testing    in: /builddir/build/BUILD/gocql-95d072f1b5bbc604ce10f41ce1c914578592ed00/_build/src
         PATH: /builddir/build/BUILD/gocql-95d072f1b5bbc604ce10f41ce1c914578592ed00/_build/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/sbin
       GOPATH: /builddir/build/BUILD/gocql-95d072f1b5bbc604ce10f41ce1c914578592ed00/_build:/usr/share/gocode
  GO111MODULE: off
      command: go test -buildmode pie -compiler gc -ldflags "-X github.com/gocql/gocql/version.commit=95d072f1b5bbc604ce10f41ce1c914578592ed00 -X github.com/gocql/gocql/version=0 -extldflags '-Wl,-z,relro -Wl,--as-needed  -Wl,-z,now -specs=/usr/lib/rpm/redhat/redhat-hardened-ld '"
      testing: github.com/gocql/gocql
github.com/gocql/gocql
PASS
ok  	github.com/gocql/gocql	2.378s
github.com/gocql/gocql/internal/lru
PASS
ok  	github.com/gocql/gocql/internal/lru	0.003s
github.com/gocql/gocql/internal/murmur
--- FAIL: TestMurmur3H1_CassandraSign (0.00s)
    murmur_test.go:65: expected -9223371632693506265 got 872418238549660192
--- FAIL: TestMurmur3H1 (0.00s)
    murmur_test.go:115: Expected h1 = -5cd6c52967134666 for data = 30313233343536373839303132333435, but was 4c3a7badc0662f83
    murmur_test.go:115: Expected h1 = -438bffdc242affb8 for data = 3031323334353637383930313233343536, but was -2fb5ccba2ddd2d82
    murmur_test.go:115: Expected h1 = 3fe5ab9837d25cdd for data = 303132333435363738393031323334353637, but was 774c27731c1fb303
    murmur_test.go:115: Expected h1 = 2d0338c1ca87d132 for data = 30313233343536373839303132333435363738, but was 43eab981281722b
    murmur_test.go:115: Expected h1 = -4761a67748c85004 for data = 3139204a616e203230333820617420333a31343a303720414d, but was -310e8a35333d267e
    murmur_test.go:115: Expected h1 = -3266b7e06116fd37 for data = 54686520717569636b2062726f776e20666f78206a756d7073206f76657220746865206c617a7920646f672e, but was -56234ade179aca85
FAIL
exit status 1
FAIL	github.com/gocql/gocql/internal/murmur	0.009s
```

--- a/internal/murmur/murmur_appengine.go
+++ b/internal/murmur/murmur_appengine.go
@@ -1,11 +1,11 @@
-// +build appengine
+// +build appengine s390x
 
 package murmur
 
 import "encoding/binary"
 
 func getBlock(data []byte, n int) (int64, int64) {
-	k1 := binary.LittleEndian.Int64(data[n*16:])
-	k2 := binary.LittleEndian.Int64(data[(n*16)+8:])
+	k1 := int64(binary.LittleEndian.Uint64(data[n*16:]))
+	k2 := int64(binary.LittleEndian.Uint64(data[(n*16)+8:]))
 	return k1, k2
 }

--- a/internal/murmur/murmur_unsafe.go
+++ b/internal/murmur/murmur_unsafe.go
@@ -1,4 +1,5 @@
 // +build !appengine
+// +build !s390x
 
 package murmur
 

